### PR TITLE
another attempt to fix #3847

### LINKED ISSUE #3847: asynchttpserver may consume unbounded memory reading headers
asynchttpserver places no restrictions on headers size or count. 

``` python
print 'GET / HTTP/1.1'
i = 0
data = 's' * 10000
while True:
    i += 1
    print 'Foo%d: ' % i, data
```

For example, script from Jester README consumes 2GB and crashes (because of ulimit) after 70 seconds (on localhost, but in practice asynchttpclient is only able to process 100 Mbit/s).

`asynchttpclient` shares the same problems.

(This is a denial of service security issue which probably shouldn't be disclosed publicly, but I don't think anyone cares about Nim security.)

### PR DESCRIPTION
**Pull request by Araq**

### CODE REVIEW

File: lib/pure/asynchttpserver.nim [hunk 1]
```diff
@@ -97,6 +100,18 @@ proc respond*(req: Request, code: HttpCode, content: string,
 
   if headers != nil:
     msg.addHeaders(headers)
+  msg.add("Content-Length: ")
```
Comment (dom96) [inline lib/pure/asynchttpserver.nim]: Now that I look at it, there should probably be a check to ensure ``headers`` don't contain a ``Content-Length`` header.

File: lib/pure/asyncnet.nim [hunk 2]
```diff
@@ -464,8 +464,7 @@ proc recvLineInto*(socket: AsyncSocket, resString: FutureVar[string],
   ## The partial line **will be lost**.
   ##
   ## The ``maxLength`` parameter determines the maximum amount of characters
-  ## that can be read before a ``ValueError`` is raised. This prevents Denial
-  ## of Service (DOS) attacks.
+  ## that can be read. ``resString`` will be truncated after that.
```
Comment (dom96) [inline lib/pure/asyncnet.nim]: Truncate it to be equal to ``maxLength`` and emphasise this in the docs to remove any doubt.

File: lib/pure/asynchttpserver.nim (hunk 1 above)
Comment (dom96) [inline lib/pure/asynchttpserver.nim]: But that's a separate issue.

File: lib/pure/asyncnet.nim (hunk 2 above)
Comment (Araq) [inline lib/pure/asyncnet.nim]: But then that's an ambiguous behaviour. What if the input's len is exactly maxLength?

Comment (dom96) [inline lib/pure/asyncnet.nim]: What if it's exactly one character more than ``maxLength``?

Comment (Araq) [inline lib/pure/asyncnet.nim]: Then it's a buffer "overflow" just like if it's exactly N characters more than maxLength.

Comment (dom96) [inline lib/pure/asyncnet.nim]: I don't see how that's ambiguous but fine.

**Pull request merged**